src/Examples/Flowsheets/HDA_flowsheet_with_costing.ipynb

```diff
@@ -0,0 +1,1598 @@
+    "    if not hasattr(unit, \"length\"):\n",
+    "        unit.length = Var(initialize=1, units=pyunits.m)\n",
+    "    if hasattr(unit, \"volume\"):  # if volume exists, set diameter from volume\n",
+    "        unit.volume_eq = Constraint(expr=unit.volume[0] == unit.length * unit.diameter**2 * 0.25 * math.pi)\n",
+    "    else:  # fix diameter directly\n",
+    "        unit.diameter.fix(0.2214 * pyunits.m)\n",
+    "    # either way, fix L/D to calculate L from D\n",
+    "    unit.L_over_D = Constraint(expr=unit.length == 3 * unit.diameter)\n",
+    "        \n",
+    "    # define vessel costing\n",
+    "    unit.costing = UnitModelCostingBlock(\n",
+    "        default={\n",
+    "            \"flowsheet_costing_block\": unit.parent_block().costing,\n",
+    "            \"costing_method\": SSLWCostingData.unit_mapping[unit_class],\n",
+    "            \"costing_method_arguments\": {\n",
+    "                \"material_type\": VesselMaterial.CarbonSteel,\n",
+    "                \"shell_thickness\": 1.25 * pyunits.inch\n",
+    "                \n",
+    "            }\n",
+    "        }\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, we will define costing for the distillation column:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# costing for column - n.fs.D101\n",
+    "\n",
+    "# define column dimensions - same volume as m.fs.F102 with L/D of 5\n",
+    "n.fs.D101.diameter = Param(initialize=0.1715, units=pyunits.m)\n",
+    "n.fs.D101.length = Param(initialize=0.8575, units=pyunits.m)\n",
+    "\n",
+    "n.fs.D101.costing = UnitModelCostingBlock(\n",
+    "        default={\n",
+    "            \"flowsheet_costing_block\": n.fs.costing,\n",
+    "            \"costing_method\": SSLWCostingData.cost_vertical_vessel,\n",
+    "            \"costing_method_arguments\": {\n",
+    "                \"material_type\": VesselMaterial.CarbonSteel,\n",
+    "                \"shell_thickness\": 1.25 * pyunits.inch,\n",
+    "                \"include_platforms_ladders\": True,\n",
+    "                \"number_of_trays\": n.fs.D101.config.number_of_trays,\n",
+    "                \"tray_material\": TrayMaterial.CarbonSteel,\n",
+    "                \"tray_type\": TrayType.Sieve\n",
+    "                \n",
+    "            }\n",
+    "        }\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Solve Flowsheet Costing Blocks\n",
+    "Now, we may solve the full flowsheet for all costing blocks:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define solver\n",
+    "from idaes.core.solvers import get_solver\n",
+    "solver = get_solver()\n",
+    "solver.options = {\"halt_on_ampl_error\": 'yes'}\n",
```

Comment:
1. You should not be setting halt on AMPL error in an example.
2. If you do need to set arguments, do it via `get_solver(options={})`